test/onnx/test_pytorch_onnx_onnxruntime.py

```diff
@@ -124,6 +125,70 @@ def _run_test(m):
             _run_test(script_model)
         _run_test(model)
 
+    def run_large_model_test(self, model, input, rtol=0.001, atol=1e-7,
+                             example_outputs=None, do_constant_folding=True,
+                             dynamic_axes=None, input_names=None, output_names=None):
+        import os
+        import tempfile
+
+        model.eval()
+        with torch.no_grad():
+            if isinstance(input, torch.Tensor):
+                input = (input,)
+            # In-place operators will update input tensor data as well.
+            # Thus inputs are replicated before every forward call.
+            input_copy = copy.deepcopy(input)
+            output = model(*input_copy)
+            if isinstance(output, torch.Tensor):
+                output = (output,)
+
+            # export the model to ONNX
+            with tempfile.TemporaryDirectory() as tmpdirname:
+                model_file_name = os.path.join(tmpdirname, 'model.onnx')
```

Comment:
Good point. The `str()` case is  an error condition for the large model format code path (as can be seen in `export.cpp`), because we need a path where to write the external binary files. Therefore, in this case, I am not sure if such a test would add much value.

Reply:
In that case, should we error out directly in _decide_large_model_format() if val_use_large_model_format=True and f is not a string (instead of waiting for the error to happen all the way in export.cpp)?

if not, a comment explaining that could be good to add here

Reply:
We cannot error out because string is empty for regular sized model test cases, e.g. test_pytorch_onnx_onnxruntime.py.